src/Diagnostics/horizontal_average.jl

```diff
@@ -42,27 +96,65 @@ function HorizontalAverage(field; frequency=nothing, interval=nothing, return_ty
     return HorizontalAverage(field, result, frequency, interval, 0.0, return_type, field.grid)
 end
 
-# Normalize a horizontal sum to get the horizontal average.
-normalize_horizontal_sum!(havg, grid) = havg.result ./= (grid.Nx * grid.Ny)
+"""
+    ZonalAverage(model, field; frequency=nothing, interval=nothing, return_type=Array)
+
+Construct a `ZonalAverage` of `field`.
+
+After the zonal average is computed it will be stored in the `result` property.
+
+The `ZonalAverage` can be used as a callable object that computes and returns the
+zonal average.
+
+A `frequency` or `interval` (or both) can be passed to indicate how often to run this
+diagnostic if it is part of `model.diagnostics`. `frequency` is a number of iterations
+while `interval` is a time interval in units of `model.clock.time`.
+
+A `return_type` can be used to specify the type returned when the `ZonalAverage` is
+used as a callable object. The default `return_type=Array` is useful when running a GPU
+model and you want to save the output to disk by passing it to an output writer.
+"""
+function ZonalAverage(field; frequency=nothing, interval=nothing, return_type=Array)
+    arch = architecture(field)
+    result = zeros(arch, field.grid, 1, total_size(parent(field))[2], total_size(parent(field))[3])
+    return ZonalAverage(field, result, frequency, interval, 0.0, return_type, field.grid)
+end
+
+# Normalize sum to get the average
+normalize_sum!(havg::VolumeAverage, grid) = havg.result ./= (grid.Nx * grid.Ny * grid.Nz)
+
+normalize_sum!(havg::HorizontalAverage, grid) = havg.result ./= (grid.Nx * grid.Ny)
+
+normalize_sum!(havg::ZonalAverage, grid) = havg.result ./= grid.Nx
 
 """
     run_diagnostic(model, havg::HorizontalAverage{NTuple{1}})
 
 Compute the horizontal average of `havg.field` and store the result in `havg.result`.
 """
-function run_diagnostic(model, havg::HorizontalAverage)
+function run_diagnostic(model, havg::AbstractAverage)
     zero_halo_regions!(parent(havg.field), model.grid)
     sum!(havg.result, parent(havg.field))
-    normalize_horizontal_sum!(havg, model.grid)
+    normalize_sum!(havg, model.grid)
     return nothing
 end
 
+function (havg::VolumeAverage{F, Nothing})(model) where F
+    run_diagnostic(model, havg)
+    return havg.result
+end
+
 function (havg::HorizontalAverage{F, Nothing})(model) where F
     run_diagnostic(model, havg)
     return havg.result
 end
 
-function (havg::HorizontalAverage)(model)
+function (havg::ZonalAverage{F, Nothing})(model) where F
+    run_diagnostic(model, havg)
+    return havg.result
+end
+
+function (havg::AbstractAverage)(model)
```

Comment:
The names of these arguments shouldnt' be `havg`, since they are not horizontal averages.